core/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala

```diff
@@ -172,16 +172,31 @@ class QualificationAppInfo(
     }
   }
 
+  private def getAppEndTime: Long = {
+    this.appEndTime match {
+      case Some(endTime) => endTime
+      case None =>
+        if (lastSQLEndTime.isEmpty && lastJobEndTime.isEmpty) {
+          0L
+        } else {
+          // estimate the app end with job or sql end times
+          val sqlEndTime = if (this.lastSQLEndTime.isEmpty) 0L else this.lastSQLEndTime.get
+          val jobEndTime = if (this.lastJobEndTime.isEmpty) 0L else lastJobEndTime.get
+          math.max(sqlEndTime, jobEndTime)
+        }
+    }
+  }
```

Comment:
There is a method that already exists that does very similar thing `QualificationAppInfo.calculateAppDuration()`

Reply:
Thanks! I was hoping we could use this function in some other scenarios. Removed it and used `appDuration + startTime` as the endTime for the app.